src/rgw/rgw_rest_s3.cc

```diff
@@ -4856,7 +4856,12 @@ int RGWHandler_REST_S3::postauth_init(optional_yield y)
   struct req_init_state *t = &s->init_state;
 
   rgw_parse_url_bucket(t->url_bucket, s->user->get_tenant(),
-		      s->bucket_tenant, s->bucket_name);
+		       s->bucket_tenant, s->bucket_name);
+  if (!s->bucket_tenant.empty() && s->bucket_name.empty()) {
+    ldpp_dout(s, 5) << "Invalid bucket URL. Bucket tenant with no bucket name."
+		    << " t->url_bucket=" << t->url_bucket << dendl;
+    return -EINVAL;
```

Comment:
i'm thinking that `ERR_METHOD_NOT_ALLOWED` -> `503 MethodNotAllowed` would be a better fit here, since this implies that we were given a request path that doesn't route to a valid rest endpoint